redbot/cogs/cleanup/cleanup.py

```diff
@@ -541,3 +541,32 @@ def check(m):
             await mass_purge(to_delete, channel)
         else:
             await slow_deletion(to_delete)
+
+    @cleanup.command(name="spam")
+    @commands.guild_only()
+    @commands.bot_has_permissions(manage_messages=True)
+    async def cleanup_spam(self, ctx: commands.Context, number: int = 50):
+        """Deletes duplicate messages in the channel from the last X messages, but keeps the original. Default 50."""
+        msgs = []
+        spam = []
+        async for msg in ctx.channel.history(limit=number, before=ctx.message, oldest_first=True):
+            if msg.attachments:
+                continue
+            c = (msg.author.id, msg.content, [e.to_dict() for e in msg.embeds])
+            if c in msgs:
+                spam.append(msg)
+            else:
+                msgs.append(c)
+
+        if len(spam) > 100:
+            cont = await self.check_100_plus(ctx, len(spam))
+            if not cont:
+                return
+
+        spam.append(ctx.message)
+        await mass_purge(spam, ctx.channel)
+        log.info(
+            "{0.author} deleted {1} spam messages in channel {0.channel}.".format(
+                ctx, len(spam) - 1
+            )
```

Comment:
Putting IDs of involved discord objects can be helpful to bot owners looking at the log so adding them in parentheses should work for that.
Also, I propose using %-formatting as it works better with various logging tooling. While we don't use any, at least currently, some bot users might be.
```suggestion
            "%s (%s) deleted %s spam messages in channel %s (%s).",
            ctx.author,
            ctx.author.id,
            len(spam) - 1,
            channel,
            channel.id,
```

Reply:
I actually just made that change! lol
I wanted to keep it with the same style